src/games/behavmixed.imp

```diff
@@ -140,44 +109,28 @@ template <class T>
 MixedBehaviorProfile<T>::MixedBehaviorProfile(const MixedStrategyProfile<T> &p_profile)
   : m_probs(p_profile.GetGame()->NumActions()),
     m_support(p_profile.GetGame()),
-    m_gameversion(p_profile.GetGame()->GetVersion()),
-    m_cacheValid(false),
-    m_realizProbs(m_support.GetGame()->NumNodes()),
-    m_beliefs(m_support.GetGame()->NumNodes()),
-    m_nvals(m_support.GetGame()->NumNodes()),
-    m_bvals(m_support.GetGame()->NumNodes()),
-    m_nodeValues(m_support.GetGame()->NumNodes(),
-		 m_support.GetGame()->NumPlayers()),
-    m_infosetValues(m_support.GetGame()->NumInfosets()),
-    m_actionValues(m_support.GetGame()->NumActions()),
-    m_regret(m_support.GetGame()->NumActions())
-{
-  m_realizProbs = (T) 0.0;
-  m_beliefs = (T) 0.0;
-  m_nodeValues = (T) 0.0;
-  m_infosetValues = (T) 0.0;
-  m_actionValues = (T) 0.0;
-  m_regret = (T) 0.0;
-
+    m_gameversion(p_profile.GetGame()->GetVersion())
+{
   ((Vector<T> &) *this).operator=((T)0);
```

Comment:
Should replace this line with

```
static_cast<Vector<T> &>(m_probs) = T(0);
```

A very good example of a line where three things can be changed, all of which used to be done because once upon a time compilers didn't like certain expressions...!